### ISSUE TITLE
what about making it cross-platform?

### ISSUE THREAD

**Issue by Helias** [OpenGOAL-Unofficial-Mods/OpenGoal-ModLauncher-dev#12]
It does not work on Linux :(
At least, the setup seems a bit odd

moreover you could add a requirements.txt file putting these packages to start `openGOALModLauncher.py`
- progressbar
- pysimpleguid
- cloudscraper

**Comment by dallmeyer** [comment]
@Zedb0T is currently working on making the launcher linux compatible 👍 

re: package requirements, we do have [this batch script](https://github.com/OpenGOAL-Unofficial-Mods/OpenGoal-ModLauncher-dev/blob/main/install-building-tools-once.bat) to help with that - we should make a linux equivalent

**Comment by Helias** [comment]
I strongly recommend to use always cross-platform scripts or any kind of file, a requirements.txt is os-independent.
Moreover, if you make it compatible for Linux will be easy to make the docker container, this will make easier the usage instead of running scripts and installing dependencies.

Thanks anyways to make this project and make it compatible with Linux, hope I will be one tester of it and hope to open some PRs to improve it at least for the usage documentation

**Comment by Zedb0T** [comment]
Yeah for now I am dropping work on this but its to a point that it should be easy to finish if someone has a better understanding of linux than I do and a better development setup.

The repo where I did most of the work is here https://github.com/OpenGOAL-Unofficial-Mods/openGOAL-terminal-launcher

The idea is basically if `platform.system()` is Linux, then use the binaries that are prepackaged rather then the windows ones that are bundled with the mods. But getting them to execute properly is proving to be a challenge

**Comment by StevenLares** [comment]
It seems that these are the tasks required to make this project Linux compatible:

- [ ] Convert windows file paths into platform agnostic paths. You can use Python's Pathlib for this. (seems to have been mostly covered by #56, currently open)
- [ ] Migrating methods from windows specific to platform agnostic methods, an example is given in #56 
- [ ] Similar to the previous task, add bash shell commands for Linux
- [ ] Remove "exe" from executable file path names if the user is on Linux. 
- [ ] On Linux, use these executables instead of the windows ones bundled from mods as @Zedb0T mentioned in this issue.

Is there anything else to add to this list?

**Comment by Zedb0T** [comment]
So, you do need to compile/build the binaries separately for linux entirely. The problem here is we were kind of lazy, and just didnt build 97% of the mods for linux. 

Most of the most are based on the OG-Mod-Base upstream, so here we should probably edit the build preset to just build for windows and linux simultaneously. This would increase build times, but remove the ability to be lazy.

**Comment by Zedb0T** [comment]
Also another thing to consider - you need to give these executables the proper settings to execute on linux. This is where I got stuck the last time I tried, but I didnt really try for more than 2 days.

**Comment by StevenLares** [comment]
Thanks for the prompt reply.

Regarding your offer with building most mods for Linux: that would help if it is not too much of a drain of your time. I am interested in adding Linux support to the launcher.

I have started reading the code base (for the launcher, and OG-Mod-Base) today as well as the faq at https://jakmods.dev/faq.
Please let me know if my understanding of the modding process is correct:

So most of the mods are based on OG-Mod-Base upstream. The mod developers then build the executable using that template. Currently, the build preset is set to Windows (as obviously most users will be playing there. This will also keep build times short). Therefore, mods are individual code bases to build rather than a patch on your existing installation, since OpenGOAL doesn't really have a modding API at the moment.

The launcher relies on JSON files (per game) that are manually maintained to track mods on GitHub. The launcher downloads the release files for the user's chosen mod (including executables, which are typically Windows).

So with the current design, you would need to build Linux versions for all (or at least most) OpenGOAL mods.

Is my understanding of applying mods correct? 
If there is something else I can read up on, please let me know.

**Comment by Zedb0T** [comment]
Yeah thats pretty much it. Mod Base has some unique functions and libraries in its binaries that are not present in the vanilla build (Mostly the sound library) so any mods that relies on these libraries must use custom binaries. Some mods you could probably get away with just grabbing vanilla and using that - but that isnt a proper practice that I'd like to support/encourage
